### ISSUE TITLE
[@typescript-eslint/ban-ts-ignore] rule not found

### ISSUE THREAD

**Issue by omarryhan** [typescript-eslint/typescript-eslint#2286]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-ts-ignore": "warn"
  }
}
```

```TS
// your repro code case
interface Foo {
  bar: string;
}

// eslint-ignore-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
const foo: Foo = {
  bar: 1
}
```

**Expected Result**

Shows a warning instead of an error when using // @ts-ignore

**Actual Result**

Shows:

`
error  Definition for rule '@typescript-eslint/ban-ts-ignore' was not found
`

On top of every .ts file

**Additional Info**

Error:

`
error  Definition for rule '@typescript-eslint/ban-ts-ignore' was not found
`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.4.15` |

**Comment by omarryhan** [comment]
It seems that it got replaced with: `@typescript-eslint/ban-ts-comment`. Not sure why though?

**Comment by omarryhan** [comment]
Just checked, there's no deprecation notice in the changelog

**Comment by bbarry** [comment]
I explicitly marking it deprecated was missed when 2.18 was released. https://github.com/typescript-eslint/typescript-eslint/issues/1854

The 3.0 release notes linked from the change log do talk about it though: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0

**Comment by bradzacher** [comment]
When upgrading to a new major release, you should always check out the release notes for every X.0.0 version you're upgrading past so you can see the breaking changes.

We are always very clear with our majors and list out every single breaking change and what it means.

@bbarry has the link for you (thanks!)
